guest/api/framework.go

```diff
@@ -0,0 +1,9 @@
+package api
+
+// ImageStateSummary provides summarized information about the state of an image.
+type ImageStateSummary struct {
+	// Size of the image
+	Size int64
+	// Used to track how many nodes have this image
+	NumNodes int
```

Comment:
minor comment:
I don't think int64 is necessary for this field. How about using int32 or a smaller variable type?